forumhelpers/FHP.js

```diff
@@ -1,3 +1,17 @@
+function getOcular(username) {
+	var ocular_pull = new XMLHttpRequest()
+	ocular_pull.open("GET", "https://my-ocular.jeffalo.net/api/user/" + username)
+	ocular_pull.send();
+	ocular_pull.onreadystatechange = function() {
+		if (ocular_pull.readyState == 4 && ocular_pull.status === 200) {
+			var ocular_data = JSON.parse(ocular_pull.responseText)
+			if (ocular_data.status) {
+				document.getElementById("ocular-status-" + username).innerHTML = ocular_data.status;
```

Comment:
do not use `innerHTML`. this opens the door for attackers to input html into an ocular status and commit xss.

Reply:
also its 2021, fetch is supported on most modern browsers now :)

Reply:
thanks!